### PR TITLE
MES-1553 - Log failed logins to custom CloudWatch log group from authoriser

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
- Some refactoring to the component tests
- Fixes issue with `package.sh` on Windows, #whocares
- Main work for this PR is in the `createLogger.ts` and `createLogger.spec.ts`

### CODE REVIEW

File: component-tests/steps/authoriseSteps.ts [hunk 1]
```diff
@@ -89,13 +80,19 @@ Given('a valid token with employee id of {string}', function (employeeId: string
...
+Given('an employee with id of {string} exists', function (employeeId: string) {
   aws.mock(
     'DynamoDB.DocumentClient', 'get', async params => ({ Item: { staffNumber: employeeId } }));
+
+  const context: AuthoriseStepsContext = this.context;
+  context.restoreDynamoDBDocumentClient = true;
 });
 
-Given('an employee with id of {string} does not exist', (employeeId: string) => {
+Given('an employee with id of {string} does not exist', function (employeeId: string) {
```
Comment (johnmorleybjss) [inline component-tests/steps/authoriseSteps.ts]: `function` required here to avoid rebinding `this`?

File: package.sh [hunk 2]
```diff
@@ -10,10 +10,10 @@ git_rev=$(git rev-parse --short HEAD)
 ignore_func_prefix="local"
 
 mkdir -p ${artefact_dir}
-functions=$(npx yaml2json serverless.yml | jq -r '.functions | keys | .[]')
+functions=$(npx yaml2json serverless.yml | jq -r '.functions | keys | .[]' | sed 's/\r\n/\n/')
```
Comment (johnmorleybjss) [inline package.sh]: Damn, what an addition. 👏 😉

File: src/functions/authoriser/framework/__tests__/createLogger.spec.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      awsSdkMock.restore('CloudWatchLogs', 'createLogStream');
+      awsSdkMock.restore('CloudWatchLogs', 'putLogEvents');
+
+      process.env.FAILED_LOGINS_CWLG_NAME = 'testLogGroupName';
+
+    });
+
+    it('should call the correct CloudWatchLogs methods', async () => {
+
+      createLogGroupSpy = sinon.stub().resolves(true);
```
Comment (johnmorleybjss) [inline src/functions/authoriser/framework/__tests__/createLogger.spec.ts]: What's the reason for Sinon here? Typemoq seems to be prevalent on the authoriser, is there something that Sinon can accomplish that Typemoq cannot?

Comment (ammarhaiderbjss) [reply src/functions/authoriser/framework/__tests__/createLogger.spec.ts]: `aws-sdk-mock` has a nice integration with sinon.

File: package.sh (hunk 2 above)
Comment (ammarhaiderbjss) [reply package.sh]: that was all @robshakespeare 🥇

File: component-tests/steps/authoriseSteps.ts (hunk 1 above)
Comment (ammarhaiderbjss) [reply component-tests/steps/authoriseSteps.ts]: again this was @robshakespeare but it looks like a `this` thing yeah

**Pull request merged**